syft/frameworks/torch/tensors/interpreters/additive_shared.py

```diff
@@ -344,6 +345,10 @@ def generate_shares(self, secret, n_workers, random_type):
         """
         random_type = torch.LongTensor if random_type == torch.int64 else torch.IntTensor
         if not isinstance(secret, random_type):
+            if "cuda" in secret.type():
```

Comment:
Q: Can't we use something like ```secret.device``` or hmm...use ```isinstance``` (I am not sure what to check against with isinstance)?

Reply:
Yes `isinstance` was a bit too narrow to capture multiple cuda types, and yes using the `device` property could also work. Initially, I thought that maybe also numpy arrays could wind up there and they don't have a `device` property (`type()` is more python native/general), but yes actually these are always tensors in these specific places, so I could also change it, what do you think?

Reply:
Hmm, I think it would make more sense to treat them as tensors (I am sorry for the late reply :( )

Reply:
Alright, I'll change it to use the `device` property then.

Reply:
Out of curiosity, what is the benefit of that compared to the current version? @gmuraru

Reply:
From my point of view if only tensors should reach this point:
* it is a safety mechanism to catch bugs
* it is also (I think) more intuitive to treat all the values as tensors since this file is in **_tensors_**/interpreters/

But, I do not have a strong opinion about this, this also looks good :+1: